server.cc

```diff
@@ -1,265 +1,98 @@
-
-  // Start accepting incoming connections
-  void run() {
-    if (!acceptor_.is_open()) return;
-    do_accept();
-  }
-
-  void do_accept() {
-    acceptor_.async_accept(socket_,
-                           std::bind(&listener::on_accept, shared_from_this(),
-                                     std::placeholders::_1));
-  }
-
-  void on_accept(boost::system::error_code ec) {
-    if (ec) {
-      fail(ec, "accept");
-    } else {
-      // Create the session and run it
-      std::make_shared<session>(std::move(socket_))->run();
-    }
-
-    // Accept another connection
-    do_accept();
-  }
-};
-
-int main(int argc, char* argv[]) {
-  auto const address = boost::asio::ip::make_address("127.0.0.1");
-  auto const port = static_cast<uint16_t>(std::atoi("8888"));
-  auto const threads = std::max<int>(1, std::atoi("16"));
-
-  // The io_context is required for all I/O
-  boost::asio::io_context ioc{threads};
-
-  // Create and launch a listening port
-  std::make_shared<listener>(ioc, tcp::endpoint{address, port})->run();
-  std::cout << "Server started..."
-            << "\n";
-
-  // Run the I/O service on the requested number of threads
-  std::vector<std::thread> v;
-  v.reserve(threads - 1);
-  for (auto i = threads - 1; i > 0; --i) v.emplace_back([&ioc] { ioc.run(); });
-  ioc.run();
-
-  return EXIT_SUCCESS;
+#include <served/plugins.hpp>
+#include <served/served.hpp>
+#include "./apis.h"
+#include "./config.h"
+#include "cJSON.h"
+
+int main(int, char const**) {
+  served::multiplexer mux;
+  mux.use_after(served::plugin::access_log);
+
+  iota_client_service_t service;
+  service.http.path = "/";
+  service.http.host = IRI_HOST;
+  service.http.port = IRI_PORT;
+  service.http.api_version = 1;
+  service.serializer_type = SR_JSON;
+  iota_client_core_init(&service);
+
+  /**
+   * @method {get} /tag/:tag Find transactions by tag
+   *
+   * @param {String} tag Must be 27 trytes long
+   *
+   * @return {String[]} hashes List of transaction hashes
+   */
+  mux.handle("/tag/{tag:[A-Z9]{27}}")
+      .get([&](served::response& res, const served::request& req) {
+        cJSON* json_obj = cJSON_CreateObject();
+        cJSON_AddStringToObject(json_obj, "command", "find_transaction_by_tag");
+        cJSON_AddStringToObject(json_obj, "tag", req.params["tag"].c_str());
+        const char* json = cJSON_PrintUnformatted(json_obj);
+        char* json_result;
+
+        api_find_transactions_by_tag(&service, json, &json_result);
+        res.set_header("content-type", "application/json");
+        res << json_result;
+
+        cJSON_Delete(json_obj);
+      });
+
+  /**
+   * @method {get} /tips Fetch all tips
+   *
+   * @return {String[]} tips List of transaction hashes
+   */
+  mux.handle("/tips").get(
+      [&](served::response& res, const served::request& req) {
+        cJSON* json_obj = cJSON_CreateObject();
+        cJSON_AddStringToObject(json_obj, "command", "get_tips");
+        cJSON_AddNumberToObject(json_obj, "opt", 2);
```

Comment:
There's no other `opt`, we can remove this column

Reply:
Corresponding api funtion still deserializes it to determine which mode to fetch. Should we also refactor it?

Reply:
Yes, thanks!